src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useTestResultsFlags/useTestResultsFlags.tsx

```diff
@@ -0,0 +1,97 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import { RepoNotFoundErrorSchema } from 'services/repo'
+import Api from 'shared/api'
+import { NetworkErrorObject, rejectNetworkError } from 'shared/api/helpers'
+
+const TestResultsFlagsSchema = z.object({
+  owner: z
+    .object({
+      repository: z.discriminatedUnion('__typename', [
+        z.object({
+          __typename: z.literal('Repository'),
+          testAnalytics: z
+            .object({
+              flags: z.array(z.string()),
+            })
+            .nullable(),
+        }),
+        RepoNotFoundErrorSchema,
+      ]),
+    })
+    .nullable(),
+})
+
+const query = `
+  query GetTestResultsFlags(
+    $owner: String!
+    $repo: String!
+    $term: String
+  ) {
+    owner(username: $owner) {
+      repository: repository(name: $repo) {
+        __typename
+        ... on Repository {
+            testAnalytics {
+             flags(term: $term)
+          }
+        }
+        ... on NotFoundError {
+          message
+        }
+      }
+    }
+  }
+  `
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+}
+
+export const useTestResultsFlags = ({ term }: { term?: string }) => {
+  const { provider, owner, repo } = useParams<URLParams>()
+
+  return useQuery({
+    queryKey: ['GetTestResultsFlags', provider, owner, repo, term],
+    queryFn: ({ signal }) =>
+      Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: {
+          provider,
+          owner,
+          repo,
+          term,
+        },
+      }).then((res) => {
+        const parsedData = TestResultsFlagsSchema.safeParse(res?.data)
+
+        if (!parsedData.success) {
+          return rejectNetworkError({
+            status: 404,
+            data: {},
+            dev: 'useTestResultsFlags - 404 Failed to parse data',
+          } satisfies NetworkErrorObject)
```

Comment:
Can we add in `error: parsedData.error` to this object (and the other unsuccessful parsing of data errors in the new hooks), the way that the helper function works is by checking the status value, and if an error is passed. You can see the underlying logic [here](https://github.com/codecov/gazebo/blob/main/src/shared/api/helpers.ts#L18-L39).

We only have an error for when parsing is unsuccessful unlike our other errors, so that is why the error value is not required, sorry for any confusion, will be working on better docs to help address these shortfalls.